metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -7960,6 +7962,14 @@ private static void startHouseKeeperService(HiveConf conf) throws Exception {
     startHouseKeeperService(conf, Class.forName("org.apache.hadoop.hive.ql.txn.AcidHouseKeeperService"));
     startHouseKeeperService(conf, Class.forName("org.apache.hadoop.hive.ql.txn.AcidCompactionHistoryService"));
     startHouseKeeperService(conf, Class.forName("org.apache.hadoop.hive.ql.txn.AcidWriteSetService"));
+
+    ThreadPool.initialize(conf);
```

Comment:
Why is this better and if it is, why only this service is converted to use the pool?
One concern is that with a pool that has > 1 thread, if some task is taking a long time (for whatever reason) you may end up the pool launching another concurrent one.  I don't think any Acid*Service are designed with that in mind.

Reply:
I just haven't gotten to the others yet.

It's better because it gives you one pool to manage the size of.  It is also a simpler design with each thread not being long running.

scheduleAtFixedRate will not result in multiple running instances of a given task.  See the javadocs.